src/components/Filters/ToggleFilter.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState } from "react";
+import { toggleFilter } from "src/components/Filters";
+import { ToggleFilterProps } from "src/components/Filters/ToggleFilter";
+import { click, render } from "src/utils/rtl";
+import { useTestIds } from "src/utils/useTestIds";
+
+describe("ToggleFilter", () => {
+  it("uses true/undefined by default", async () => {
+    // Given a default boolean filter
+    const r = await render(<TestFilter toggleFilter={{}} />);
+    // It's initially empty
+    expect(r.filter_filter()).not.toBeChecked();
+    expect(r.value()).toHaveTextContent("undefined");
+    // And when they click it, we turn true
+    click(r.filter_filter());
+    expect(r.filter_filter()).toBeChecked();
+    expect(r.value()).toHaveTextContent("true");
+    // And when they click it again, we go back to undefined
+    click(r.filter_filter());
+    expect(r.filter_filter()).not.toBeChecked();
+    expect(r.value()).toHaveTextContent("undefined");
+  });
+
+  it("can use false/undefined when onValue is set", async () => {
+    // Given a default boolean filter
+    const r = await render(<TestFilter toggleFilter={{ onValue: false }} />);
+    // It's initially empty
+    expect(r.filter_filter()).not.toBeChecked();
+    expect(r.value()).toHaveTextContent("undefined");
+    // And when they click it, we turn true
+    click(r.filter_filter());
+    expect(r.filter_filter()).toBeChecked();
+    expect(r.value()).toHaveTextContent("false");
+    // And when they click it again, we go back to undefined
+    click(r.filter_filter());
+    expect(r.filter_filter()).not.toBeChecked();
+    expect(r.value()).toHaveTextContent("undefined");
+  });
+
+  it("can use true/false when offValue is set", async () => {
+    // Given a default boolean filter
+    const r = await render(<TestFilter toggleFilter={{ offValue: false }} />);
+    // It's initially empty
+    expect(r.filter_filter()).not.toBeChecked();
+    expect(r.value()).toHaveTextContent("false");
+    // And when they click it, we turn true
+    click(r.filter_filter());
+    expect(r.filter_filter()).toBeChecked();
+    expect(r.value()).toHaveTextContent("true");
+    // And when they click it again, we go back to false
+    click(r.filter_filter());
+    expect(r.filter_filter()).not.toBeChecked();
+    expect(r.value()).toHaveTextContent("false");
+  });
+
+  it("can use foo/bar when onValue/offValue are both set", async () => {
+    // Given a default boolean filter
+    const r = await render(<TestFilter toggleFilter={{ onValue: "foo", offValue: "bar" }} />);
+    // It's initially empty
+    expect(r.filter_filter()).not.toBeChecked();
```

Comment:
**question/suggestion:** Can we have another test to validate that this Toggle _can_ be initially checked?

Reply:
Yep, added.